File: backend/model/leaderboardData.go
package model

type LeaderboardDataset struct {
	Person      Person      `json:"person"`
	LeagueEntry LeagueEntry `json:"league_entry"`
	Rank        int         `json:"rank"`
	Value       interface{} `json:"value"` // BSM returns both strings and integers for different endpoints
}

type LeaderboardData struct {
	League        LeagueGroup          `json:"league"`
	StatsCategory string               `json:"stats_category"`
	Data          []LeaderboardDataset `json:"data"`
}

type LeaderboardSummary struct {
	LeagueID  string            `json:"league_id"`
	StatsType string            `json:"stats_type"`
	Data      []LeaderboardData `json:"data"`
}

File: src/lib/model/LeaderboardData.ts
import type {LeagueEntry, LeagueGroup, Person} from "bsm.js";

// this data has a custom structure and is therefore project-specific and not in bsm.js
export interface LeaderboardDataset {
  person: Person;
  league_entry: LeagueEntry;
  rank: number;
  value: string | number; // Value can be a string or a number depending on the endpoint
}

export interface LeaderboardData {
  league: LeagueGroup;
  stats_category: string;
  data: LeaderboardDataset[];
}

export interface LeaderboardSummary {
  league_id: string;
  stats_type: string;
  data: LeaderboardData[];
}
